src/Cluster.jl

```diff
@@ -0,0 +1,57 @@
+using LinearAlgebra
+using Arpack
+
+function optimal_num_clusters(eigenvalues::Vector{Float64})
+    # find index where the gap between successive eigenvalues is maximized
+    gaps = diff(eigenvalues)
+    _, index = findmax(gaps)
+    # add 1 due to 0-indexing to get optimal number of clusters 
+    opt_num_clusters = index + 1 
+    return opt_num_clusters
+end
+
+function pcca(transition_matrix::Matrix{Float64})
+    n_states = size(transition_matrix, 1)
+    
+    # verify transition_matrix is row stochastic (sums to 1)
+    for i in 1:n_states
+        @assert abs(sum(transition_matrix[i, :]) - 1) < 1e-10 "Row $i of transition matrix doesn't sum to 1"
```

Comment:
It's better to check conditions on the input explicitly, and throw `ArgumentError` when violated.  `@assert` statements can be removed by the compiler.

One good way to do it is like the function `augmentedRateMatrix`

Reply:
Thanks for the tip! I'm already quite ahead of this, but I have changed asserts in my follow up PR that I'm preparing now to throw errors.